apps/src/javalab/JavalabView.jsx

```diff
@@ -140,6 +141,26 @@ class JavalabView extends React.Component {
     );
   };
 
+  onCaptchaVerify = () => {
+    const {isRunning, isTesting, onRun, onTest} = this.props;
+    if (isRunning) {
+      onRun();
+    }
+    if (isTesting) {
+      onTest();
+    }
+  };
+
+  onCaptchaCancel = () => {
+    const {isRunning, isTesting, setIsRunning, setIsTesting} = this.props;
```

Comment:
This is...not beautiful, but I feel like our run/test state is unnecessarily complicated? Should we make this an enum or something?

Reply:
hmm yeah if you wanted to we could change from flags to a single enum `RUNNING, TESTING, OFF` or something like that? But that doesn't have to happen as part of this work, as it probably requires changing a few files

Reply:
Similarly, if we see ourselves doing this in a couple places, it'd be worth setting up some sort of (locally) global setter function so we're guaranteed to never have more than one of these states set to true at a time.

Reply:
Yeah agree with both of these points -- I just added an item here to look into this more:

https://codedotorg.atlassian.net/browse/SL-852